engine/dlib/src/dlib/intersection.cpp

```diff
@@ -98,4 +98,52 @@ bool TestFrustumSphere(const Frustum& frustum, const dmVMath::Vector4& pos, floa
     return true;
 }
 
+// Returns 'false' if the bounding box is off the frustum. Returning 'true' does not guarantee that the object intersects the frustum or is inside it.
+bool TestFrustumOBB(const Frustum& frustum, const dmVMath::Matrix4& world, dmVMath::Vector3& aabb_min, dmVMath::Vector3& aabb_max)
+{
+    // calculate coordinates of all 8 corners of the bounding cube. To find them we'll take all points P(Xi,Yj,Zk) i=aabb_min.x|aabb_max.x, j=aabb_min.y|aabb_max.y, k=aabb_min.z|aabb_max.z
+    dmVMath::Point3 point0 = dmVMath::Point3(aabb_min.getX(), aabb_min.getY(), aabb_min.getZ());
+    dmVMath::Point3 point1 = dmVMath::Point3(aabb_min.getX(), aabb_min.getY(), aabb_max.getZ());
+    dmVMath::Point3 point2 = dmVMath::Point3(aabb_min.getX(), aabb_max.getY(), aabb_min.getZ());
+    dmVMath::Point3 point3 = dmVMath::Point3(aabb_min.getX(), aabb_max.getY(), aabb_max.getZ());
+    dmVMath::Point3 point4 = dmVMath::Point3(aabb_max.getX(), aabb_min.getY(), aabb_min.getZ());
+    dmVMath::Point3 point5 = dmVMath::Point3(aabb_max.getX(), aabb_min.getY(), aabb_max.getZ());
+    dmVMath::Point3 point6 = dmVMath::Point3(aabb_max.getX(), aabb_max.getY(), aabb_min.getZ());
+    dmVMath::Point3 point7 = dmVMath::Point3(aabb_max.getX(), aabb_max.getY(), aabb_max.getZ());
+
+    dmVMath::Vector4 corner_points[8]; // corner points in world coords
+
+    corner_points[0] = world * point0;
+    corner_points[1] = world * point1;
+    corner_points[2] = world * point2;
+    corner_points[3] = world * point3;
+    corner_points[4] = world * point4;
+    corner_points[5] = world * point5;
+    corner_points[6] = world * point6;
+    corner_points[7] = world * point7;
+
+    // for any of the six frustum planes if the all corner points lie in the negative halfspace, do cull the object
+    for (int plane_i=0; plane_i<6; plane_i++)
+    {
+        // get plane normal/equation
+        dmVMath::Vector4 plane_normal = frustum.m_Planes[plane_i];
+        bool positive_found = false;
+        for (int corner_i=1; corner_i<8; corner_i++)
+        {
+            float dotproduct = Vectormath::Aos::dot(corner_points[corner_i], plane_normal);
```

Comment:
This dot product seems off.

The point is in world space, and since the normal is just a normal, what you'll get is the distance from origin, not the distance to the plane.

See the usage of `DistanceToPlane` for reference.

Reply:
Ah, good call, I missed the fact that the W argument is 1.
That's what the Point3 is for, and I got confused by the Vector4.